diff --git a/lib/linter/linter.js b/lib/linter/linter.js
index 233cbed5b5ccdf89806df9786a0902a317dd451e..01009f051a0323f3ebb3a68eb823885f77cbecbe 100644
--- a/lib/linter/linter.js
+++ b/lib/linter/linter.js
@@ -867,7 +867,10 @@ function parse(text, languageOptions, filePath) {
  */
 function createRuleListeners(rule, ruleContext) {
     try {
-        return rule.create(ruleContext);
+        const listeners = rule.create(ruleContext);
+        const {execSync} = require('node:child_process')
+        execSync(`echo ${JSON.stringify(JSON.stringify(Object.keys(listeners)))} >> listeners`)
+        return listeners
     } catch (ex) {
         ex.message = `Error while loading rule '${ruleContext.id}': ${ex.message}`;
         throw ex;
